libs/onboarding-sdk-product/src/main/java/it/pagopa/selfcare/product/entity/Product.java

```diff
@@ -347,4 +357,24 @@ && getInstitutionContractMappings().containsKey(institutionType)) {
     }
     return contractTemplate;
   }
+
+  /**
+   * This method returns ContractTemplate associate with a specific InstitutionType for UserAggregatorContract.
+   * In case none InstitutionType exists on contractMapping, it returns the default ContractTemplate.
+   *
+   * @param institutionType InstitutionType
+   * @return ContractTemplate
+   */
+  public ContractTemplate getUserAggregatorContractTemplate(String institutionType) {
+    ContractTemplate userAggregatorContractTemplate = new ContractTemplate();
+    if (Objects.nonNull(getUserAggregatorContractMappings())) {
+      if (Objects.nonNull(institutionType)
+              && getUserAggregatorContractMappings().containsKey(institutionType)) {
+        userAggregatorContractTemplate = getUserAggregatorContractMappings().get(institutionType);
+      } else if (getUserAggregatorContractMappings().containsKey(CONTRACT_TYPE_DEFAULT)) {
+        userAggregatorContractTemplate = getUserAggregatorContractMappings().get(CONTRACT_TYPE_DEFAULT);
+      }
+    }
+    return userAggregatorContractTemplate;
+  }
```

Comment:
you couldtWrite tests to verify the behavior of the getUserAggregatorContractTemplate method under various scenarios. for ex:

```
@Test
public void testGetUserAggregatorContractTemplate_WithValidInstitutionType() {
    // Setup
    Product product = new Product();
    Map<String, ContractTemplate> mappings = new HashMap<>();
    mappings.put("validType", new ContractTemplate(/* parameters */));
    product.setUserAggregatorContractMappings(mappings);
    
    // Execute
    ContractTemplate result = product.getUserAggregatorContractTemplate("validType");
    
    // Verify
    assertNotNull(result);
    assertEquals(mappings.get("validType"), result);
}

@Test
public void testGetUserAggregatorContractTemplate_WithDefault() {
    // Setup
    Product product = new Product();
    Map<String, ContractTemplate> mappings = new HashMap<>();
    mappings.put("default", new ContractTemplate(/* parameters */));
    product.setUserAggregatorContractMappings(mappings);
    
    // Execute
    ContractTemplate result = product.getUserAggregatorContractTemplate("unknownType");
    
    // Verify
    assertNotNull(result);
    assertEquals(mappings.get("default"), result);
}

@Test
public void testGetUserAggregatorContractTemplate_NoMappings() {
    // Setup
    Product product = new Product();
    product.setUserAggregatorContractMappings(null);
    
    // Execute
    ContractTemplate result = product.getUserAggregatorContractTemplate("anyType");
    
    // Verify
    assertNotNull(result); // Depending on desired behavior
    // Further assertions based on implementation
}
```